73609

SAM19991 opened the pull request.

SAM19991 committed:
73609

iBerbasov commented on src/main/java/school/faang/bjs273609/Main.java:
```diff
@@ -0,0 +1,38 @@
+                executor.submit(() -> {
+                    while (!game.isGameOver()) {
+                        boolean winner = Math.random() < 0.7;
+                        boolean loose = Math.random() < 0.1;
```
Имена переменных для boolean лучше начинать с is, например: isWin и isLoose

iBerbasov commented on src/main/java/school/faang/bjs273609/Main.java:
```diff
@@ -0,0 +1,38 @@
+        IntStream.range(MIN_RANGE, MAX_RANGE).forEach(i ->
+                executor.submit(() -> {
+                    while (!game.isGameOver()) {
+                        boolean winner = Math.random() < 0.7;
```
Подобные магические числа лучше выносить в константы с информативными именами, например CHANCE_ADD_POINT CHANCE_LOST_LIVE

iBerbasov commented on src/main/java/school/faang/bjs273609/Game.java:
```diff
@@ -0,0 +1,40 @@
+public class Game {
+    private static final int MAX_LIVES = 10;
+    private final Object lockScore = new Object();
+    private final Object lockLives = new Object();
```
Локов объявлено два, а используется только один
По условию задачи надо отдельно синхронизировать получение очков и потерю жизней - сделай два synchronized блока
Только не забудь, что если проверка gameOver останется вне этих блоков - то будет возникать состояние гонки

iBerbasov commented on src/main/java/school/faang/bjs273609/Game.java:
```diff
@@ -0,0 +1,40 @@
+                System.out.printf("Score was increased by %d (%s)\n", score, Thread.currentThread().getName());
+            }
+            if (loose) {
+                lives++;
```
У тебя конечно тоже рабочий вариант, но почему решил сделать так?)
Обычно жизни отнимаются и при нуле - game over ))

SAM19991 replied:
Там написано "lives — общее количество потерянных жизней всеми участниками." , то-есть я понял так, что нужно сложить общее количество потерянных жизней. Я иногда не полностью понимаю условия задач , в связи с чем и получаются такие решения.

iBerbasov replied:
Внимательно вчитываешься в задание - это полезный навык ))
Я и не обратил внимание, после lives решил, что мне всё понятно. А оказывается подковырка есть

SAM19991 committed:
Moved magic numbers into constants.
Renamed flags to isWin/isLose.
Split synchronized blocks for score and lives updates.
